### PR TITLE
Query Meos Master for Sidecar seeds

### PR DESCRIPTION
**Pull request by relistan**
If the `EXECUTOR_SEED_SIDECAR` env var is set, this will cause the executor to query the Mesos master for the list of Sidecar seed hosts. The list is all known Mesos nodes at the moment. We could cut that back by adding another setting if needed.

This gets the list by first calling the local node over http to get the master hostname, and then calling the `/slaves` endpoint on the master and collecting all the hostnames.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -157,6 +164,14 @@ with `EXECUTOR_`.
    they show up in the Mesos logs. `LogsSince` is how far back in time we
    reach to get these logs.
 
+ * **ForceCpuLimit**: Should we enfoce the CPU limits in the request using
+   Docker's CPU using cgroups?
+
+ * **ForceMeoryLimit**: Should we enfoce the memory limits in the request using
+   Docker's CPU using cgroups?
```
Comment (mihaitodor) [inline README.md]: Docker's memory?

File: callbacks.go [hunk 2]
```diff
@@ -136,6 +152,105 @@ func (exec *sidecarExecutor) LaunchTask(driver executor.ExecutorDriver, taskInfo
...
+
+	// Let's find out the Mesos master's hostname
+	resp, err := exec.fetcher.Get(localEndpoint)
+	defer func() {
+		if resp != nil {
+			resp.Body.Close()
+		}
+	}()
+
+	if err != nil {
```
Comment (mihaitodor) [inline callbacks.go]: I think this err check should go before the defer above. I think Body can be nil if err != nil.

File: callbacks.go [hunk 3]
```diff
@@ -136,6 +152,105 @@ func (exec *sidecarExecutor) LaunchTask(driver executor.ExecutorDriver, taskInfo
...
+
+	err = json.Unmarshal(body, &localStruct)
+	if err != nil {
+		return "", fmt.Errorf("Error parsing response body from Mesos worker! '%s'", err)
+	}
+
+	return localStruct.MasterHostname, nil
+}
+
+// Used only in getWorkerHostnames
```
Comment (mihaitodor) [inline callbacks.go]: then why not declare it in the body of the function?

File: callbacks.go [hunk 4]
```diff
@@ -136,6 +152,105 @@ func (exec *sidecarExecutor) LaunchTask(driver executor.ExecutorDriver, taskInfo
...
+
+	masterStruct := struct {
+		Slaves []workersStruct `json:"slaves"`
+	}{}
+
+	// Let's find out the Mesos master's hostname
+	resp, err := exec.fetcher.Get(masterEndpoint)
+	defer func() {
+		if resp != nil {
+			resp.Body.Close()
```
Comment (mihaitodor) [inline callbacks.go]: Same issue as before.

File: callbacks.go (hunk 3 above)
Comment (relistan) [reply callbacks.go]: haha, I guess that works now :)

**Pull request merged**